Doesn't compile with GCC 15

coordgenlibs 3.0.2
gcc 15.2.0

```
[ 23%] Building CXX object CMakeFiles/coordgen.dir/CoordgenTemplates.cpp.o
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/vector:67,
                 from /build/source/CoordgenMinimizer.h:13,
                 from /build/source/CoordgenMinimizer.cpp:6:
In function '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<CoordgenFragmentDOF* const*, vector<CoordgenFragmentDOF*> >; _ForwardIterator = CoordgenFragmentDOF**]',
    inlined from '_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<CoordgenFragmentDOF* const*, vector<CoordgenFragmentDOF*> >; _Sentinel = __gnu_cxx::__normal_iterator<CoordgenFragmentDOF* const*, vector<CoordgenFragmentDOF*> >; _ForwardIterator = CoordgenFragmentDOF**; _Tp = CoordgenFragmentDOF*]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:637:37,
    inlined from 'std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = CoordgenFragmentDOF*; _Alloc = std::allocator<CoordgenFragmentDOF*>]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_vector.h:636:31,
    inlined from 'std::vector<std::vector<CoordgenFragmentDOF*> > CoordgenMinimizer::buildTuplesOfDofs(const std::vector<CoordgenFragmentDOF*>&, unsigned int) const' at /build/source/CoordgenMinimizer.cpp:1093:37:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:273:31: error: 'void* __builtin_memcpy(void*, const void*, long unsigned int)' accessing 9223372036854775800 bytes at offsets 0 and 0 overlaps 9223372036854775793 bytes at offset 7 [-Werror=restrict]
  273 |               __builtin_memcpy(std::__niter_base(__result),
      |               ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  274 |                                std::__niter_base(__first),
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  275 |                                __n * sizeof(_ValT));
      |                                ~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/CoordgenMinimizer.cpp
+++ b/CoordgenMinimizer.cpp
@@ -1086,7 +1086,7 @@ CoordgenMinimizer::buildTuplesOfDofs(const vector<CoordgenFragmentDOF*>& dofs,
     for (unsigned int i = 1; i < order; ++i) {
         lastOrderVector = growingVector;
         growingVector.clear();
-        for (auto lastOrderTuple : lastOrderVector) {
+        for (const auto& lastOrderTuple : lastOrderVector) {
             bool copy = false;
             for (auto dof : dofs) {
                 if (copy) {

